lib/kafka/consumer.rb

```diff
@@ -288,6 +288,10 @@ def mark_message_as_processed(message)
       @offset_manager.mark_as_processed(message.topic, message.partition, message.offset)
     end
 
+    def send_heartbeat_if_necessary
```

Comment:
I think we should probably simplify the API in order to avoid confusion – maybe it would make more sense to only allow clients to force a heartbeat manually?